presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java

```diff
@@ -321,13 +323,35 @@ public void onSuccess(PagesResponse result)
                         }
 
                         if (result.getToken() == token) {
+                            shouldAcknowledge = true;
                             pages = result.getPages();
                             token = result.getNextToken();
                         }
                         else {
                             pages = ImmutableList.of();
                         }
                     }
+
+                    if (shouldAcknowledge) {
+                        // Acknowledge token without handling the response.
+                        // The next request will also make sure the token is acknowledged.
+                        // This is to fast release the pages on the buffer side.
+                        URI uri = HttpUriBuilder.uriBuilderFrom(location).appendPath(String.valueOf(result.getToken())).appendPath(ACKNOWLEDGE_PATH).build();
+                        httpClient.executeAsync(prepareGet().setUri(uri).build(), new ResponseHandler<Response, RuntimeException>()
```

Comment:
I think you can make this `new ResponseHandler<Void, RuntimeException>` and return null from the `handle` method

Reply:
I'd use `createStatusResponseHandler()` here